src/Behat/Behat/Context/Environment/InitializedContextEnvironment.php

```diff
@@ -115,8 +114,6 @@ public function getContexts()
     /**
      * Returns registered context by its class name.
      *
-     * @template T of Context
```

Comment:
This was declared at the class level

Reply:
this is totally wrong. A class-level template or a method-level template are totally different behaviors.

The old signature was saying that anytime you pass a class string to that `getContext` class, you get an instance of the requested class.
the new signature means that you have to pass a class name being a subtype of the generic type of the environment instance (which cannot have any proper type more precise than `Context` as its generic type due to the way Behat works) and you get an instance of that generic type (i.e. an instance of `Context`), with no guarantee that it is an instance of the requested class.